### ISSUE TITLE
Styling: Fix collapsed site header hamburger menu

### ISSUE THREAD

**Issue by seanmiller26** [zooniverse/front-end-monorepo#4207]
## Is your feature request related to a problem? Please describe.
The hamburger menu has incorrect spacing and an unnecessary dropdown icon.

![Screen Shot 2023-01-27 at 11 55 29 AM](https://user-images.githubusercontent.com/89043489/215160715-849606c4-f447-433f-b227-ef08c863a1ce.png)

## Describe the solution you'd like
Fix the spacing to match the other icons (20px). It currently has added padding of 20px on each side. Remove the dropdown icon. It is implied that the hamburger menu will expand when clicked. Add a 30px margin on the right, similar to the Zooni logo on the left of the navbar.

<img width="683" alt="Screen Shot 2023-01-27 at 12 05 36 PM" src="https://user-images.githubusercontent.com/89043489/215162371-797b8ae9-4fb1-45ed-a8da-fd856cc26e5d.png">
<img width="739" alt="Screen Shot 2023-01-27 at 12 09 01 PM" src="https://user-images.githubusercontent.com/89043489/215162389-f3714a64-736e-44d3-b043-0f6e297987a8.png">

Once clicked, remove the dropdown icon and retain the initial positioning of the hamburger menu icon. The color should be #FFFFFF. The highlight should display as shown with 10px to the left of the hamburger icon, if possible.

<img width="407" alt="Screen Shot 2023-01-27 at 12 45 03 PM" src="https://user-images.githubusercontent.com/89043489/215169345-4a3ce935-f4d5-4cb2-bbf3-2b889c8b8a1e.png">

**Comment by eatyourgreens** [comment]
D’oh! This is my fault, from when I added admin mode. The button is showing the default Grommet drop icon because we don’t explicitly set the `icon` prop to false.

**Comment by eatyourgreens** [comment]
Easy fix though.
